### ISSUE TITLE
Bookmark timestamps on the songs' timeline UI, so that user can get back quickly to banger moments :100: 

### ISSUE THREAD

**Issue by sedha344** [OxygenCobalt/Auxio#846]
### Description

For people who need instant motivation because they're in peak action, or simply to mark song segments for some editing later,
would be good if the app created a map/attributes, from each song, to one or several timestamps, and provide handles to these timestamps on the UI timeline.

Would be really empowering for the people if y'all achieve it :heart: 

(Bonus point if there's one parameter to vary from how long before the banger occurs, the user wants to land at. Basically something like "Get me __ seconds before each added banger happens")

### Other implementations

Poweramp trial version apk -> Library list options -> Tick bookmarks -> Then on poweramp main screen, click 3dots or hold album artpic -> Press bookmarks -> Then user has access to bookmarks in their library.

The bummer of poweramp tho is that, besides being a ~~"for life"~~ paid app, it has to have network access for license checks on a frequent basis : so big nono for privacy, and makes it impossible to carry an offgrid phone in place of an ipod or some dedicated mp3 player (and other purposes).

### Benefit

Humanity will get a lot more HEROES :fist_raised: 

Thanks my heroes :pray: :ok

### Duplicates

- [X] I have checked this feature request for duplicates.
- [X] I have checked that this feature is not implemented in the [lastest version](https://github.com/OxygenCobalt/Auxio/releases).
- [X] I have checked the [Why Are These Features Missing?](https://github.com/OxygenCobalt/Auxio/wiki/Why-Are-These-Features-Missing%3F) page.
- [X] I agree to the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).

**Comment by OxygenCobalt** [comment]
Dupe of what was proposed in #762 originally.
- There is no semantic space left in the playback UI to properly integrate a new bookmark button, and it will unnecessarily take up space elsewhere.
- It's extremely niche, considering how poweramp nests it within several layers of settings and then hides it inside the three dot menu.

I simply cannot maintain something like this in my free time given all the places it will touch within Auxio's code and the small amount of people that use it. I personally won't, so it'll just break over time or break something else.

**Comment by sedha344** [comment]
Just read #762 , gotta admit idk how the internals work, but i'm not necessarily asking for such a demanding task of physically splitting files. I was more thinking about a "go to" (this timestamp) handle... I have 3minute long songs in mind, not 2hour concerts as in previous post

> It's extremely niche, considering how poweramp nests it within several layers of settings and then hides it inside the three dot menu.

I'm convinced it actually wouldn't be that niche, if poweramp author ever advertised the feature, he just never did it. One had to ask them personally, and one taught me this personally, that's how I'm aware of it.
I'm sure whole humanity would be hyped up by such feature, but I understand that a one-man army can't be on all the battlefronts, so respect anyway :pray: 

Would appreciate if you posted a link to some draft/branch tho, in case you have free time and ever try implementing this finicky feature someday :pray: I'd then be willing to live with this build forever without updating, for it not to break.

(I'm not sure whether semantic space means storage space the app takes, but i'm willing to settle for some initial download/homecalling, in case the source code happened to exceed allowed apk size of some app repositories)

Thanks anyway, keep up the good work my Hero :pray: :medal_sports: :heart: :headphones:

**Comment by OxygenCobalt** [comment]
> Just read https://github.com/OxygenCobalt/Auxio/issues/762 , gotta admit idk how the internals work, but i'm not necessarily asking for such a demanding task of physically splitting files. I was more thinking about a "go to" (this timestamp) handle... I have 3minute long songs in mind, not 2hour concerts as in previous post

It's still the same UI they were originally asking for, a way to bookmark and visit a particular moment. I rejected the bookmarking system but accepted cuesheets since it served that user's purpose.

> I'm convinced it actually wouldn't be that niche, if poweramp author ever advertised the feature, he just never did it. One had to ask them personally, and one taught me this personally, that's how I'm aware of it.
> I'm sure whole humanity would be hyped up by such feature, but I understand that a one-man army can't be on all the battlefronts, so respect anyway 🙏

That to me implies it is quite niche, given that poweramp's author has the incentive to publicize compelling features for sales and would absolutely share about the bookmarking feature rather than hiding it. Since the functionality is low-key, I have the feeling it's quite niche.

> (I'm not sure whether semantic space means storage space the app takes, but i'm willing to settle for some initial download/homecalling, in case the source code happened to exceed allowed apk size of some app repositories)

Basically, 'semantic space' how much I can fit into a UI without a user seeing it as cluttered or otherwise unappealing. I maxed out this metric in the playback UI at roughly Auxio 2.0.0, and can't add much else.
